python/tvm/driver/tvmc/main.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+TVMC - TVM driver command-line interface
+"""
+import argparse
+import logging
+import sys
+
+import pkg_resources
+
+from tvm.driver.tvmc.common import TVMCException
+
+
+def _main(argv):
+    """ TVM command line interface. """
+
+    parser = argparse.ArgumentParser(
+        prog='tvmc',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="tvm compiler driver",
+        epilog=__doc__,
+    )
+    parser.add_argument(
+        "-v", "--verbose", action="count", default=0, help="increase verbosity"
+    )
+    parser.add_argument(
+        "--version", action="store_true", help="print the version and exit"
+    )
+
+    # TODO: subparsers will come in follow-up patches (@leandron)
+    _ = parser.add_subparsers(title="commands")
+
+    args = parser.parse_args(argv)
+    if args.verbose > 4:
+        args.verbose = 4
+
+    logging.getLogger().setLevel(40 - args.verbose * 10)
+
+    if args.version:
+        version = pkg_resources.get_distribution("tvm").version
+        sys.stdout.write("%s\n" % version)
+        return 0
+
+    if not hasattr(args, "func"):
+        parser.error("missing subcommand")
```

Comment:
Since you don't have any subparser in this PR, I guess you will require each subparser to define an argument `func` and set its default value to the entry function. IIUC, this is a pure internal error and the error message has to be improved.

Reply:
Sure, I'll add a small placeholder example, so that in the specific (real) subparsers we can review the whole funcionality.

Reply:
Reading that again, it seems most cases will be covered by `argparse` and that should really be an assert, because lacking a `func` only is expected to happen during development.

Reply:
make sense.